### ISSUE TITLE
Daemon log output subsystem names truncated

### ISSUE THREAD

**Issue by JesseWeinstein** [ipfs/go-ipfs#3161]
Version: 0.4.3-rc3
Type: bug
Area: commands
Priority: P0

The log output printed by the daemon (not the event log output produced by `ipfs log tail`) truncates the subsystem names (e.g. core/commands gets turned into core/comma) which are not accepted by `ipfs log level`, making it hard to figure out how to filter them out, and generally confusing the user.

I see three possible fixes:
1. Expand the available space for subsystems in the output to fit the longest subsystem name, and add checks to prevent the creation of longer subsystem names. 
2. Shorten the existing subsystem names to fit the current available space.
3. Add support to `ipfs log level` to accept unique prefixes for subsystem names.

These could be combined, if wanted.
I am interested in making a PR for this, but wanted to discuss it first to see what solution is preferred.

**Comment by whyrusleeping** [comment]
I think adding support for matching on the prefixes would be a good idea. And if theres a collision we can print an error like "ambiguous reference, did you mean ABCX or ABCY?"

**Comment by Stebalien** [comment]
Fixed?

**Comment by JesseWeinstein** [comment]
@Stebalien I have now confirmed that this is **NOT** fixed in the current (0.4.23) version. Please re-open it.

You can verify it isn't fixed by running:
`ipfs daemon &` 
waiting for it to start up, then running:
`ipfs log level all debug`
and looking at the third field in the log output, for example:
`17:59:00.241 DEBUG blockservi: blockservice is shutting down... blockservice.go:321`

I also confirmed that the `ipfs log level` does not support prefixes.

**Comment by Stebalien** [comment]
Ah, _that_ is definitely fixed in master. We've switched to zap as the backend.

```
2020-02-21T13:56:30.982-0500	DEBUG	blockservice	go-blockservice@v0.1.2/blockservice.go:335	blockservice is shutting down...
```
